### ISSUE TITLE
LOGIN_ERROR with null error dispatched before all LOGIN actions

### ISSUE THREAD

**Issue by el1f** [prescottprue/react-redux-firebase#690]
**Do you want to request a *feature* or report a *bug*?**  

bug

**What is the current behavior?**

Whenever `this.props.firebase.login()` is called a `LOGIN_ERROR` action is dispatched with a single `null` item as the content of the `state.firebase.errors` array before dispatching the `LOGIN` action with the correct data.
![image](https://user-images.githubusercontent.com/10723208/56801958-11989d00-681f-11e9-8ea2-471bd31c337d.png)

**If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem via [codesandbox](https://codesandbox.io/) or similar.**

**What is the expected behavior?**

I'd expect the LOGIN action to be dispatching without needing the LOGIN_ERROR action as it mkes it harder to use the `isEmpty` and `isLoaded` properties of the profile.

**Which versions of dependencies, and which browser and OS are affected by this issue? Did this work in previous versions or setups?**

- react-redux-firebase `^2.2.6`
- Chrome on MacOS `73.0.3683.103`

**Comment by el1f** [comment]
Probably a duplicate of #388, tried the possible solutions proposed in that issue but it didn't seem to help in any meaningful way. Also, when multiple logins fail the `errors` array is filled by consequential pushes of `null` elements. Is this the intended behaviour?

**Comment by prescottprue** [comment]
@el1f Open to a change in the reducer logic, what is it specifically that makes it "harder to use `isEmpty` and `isLoaded`"? The `LOGIN_ERROR` action dispatch with null is to clear any existing errors, but definitely not clear naming, so maybe a new action name?

**Comment by el1f** [comment]
That would be a lot of help already! I'm starting to dig deeper into the specific issue I had and I've already unearthead details that make me strongly think that I am actually mishandling my specific use case. Still this is a collateral thing that obviously misled me into thinking that my issues were caused by the lib not behaving as it should. I would definitely expect `LOGIN_ERROR` to be there to actually only deal with adding errors, would definitely go for adding an extra `LOGIN_ERROR_CLEAR` action or something similar to get rid of the errors.

Also, this could be OT but the clearing is not always really reliable as noted above as I've realized that when we have several consequential errors it ends up with an ever increasingly big array of `null` which I definitly don't expect to be the desired situation.
